stepactions/reduce-snapshot/reduce-snapshot.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: StepAction
+metadata:
+  name: reduce-snapshot-action
+spec:
+  image: quay.io/konflux-ci/appstudio-utils:ab6b0b8e40e440158e7288c73aff1cf83a2cc8a9@sha256:24179f0efd06c65d16868c2d7eb82573cce8e43533de6cea14fec3b7446e0b14
+  params:
+    - name: IMAGES
+      type: string
+    - name: SINGLE_COMPONENT
+      type: string
+    - name: CR_TO_QUERY
+      type: string
+    - name: CR_TO_QUERY_NAMESPACE
+      type: string
+      default: ""
+    - name: SNAPSHOT_PATH
+      type: string
+  env:
+    - name: SNAPSHOT
+      value: $(params.IMAGES)
+    - name: SINGLE_COMPONENT
+      value: $(params.SINGLE_COMPONENT)
+    - name: CR_TO_QUERY
+      value: $(params.CR_TO_QUERY)
+    - name: CR_TO_QUERY_NAMESPACE
+      value: $(params.CR_TO_QUERY_NAMESPACE)
+    - name: SNAPSHOT_PATH
+      value: $(params.SNAPSHOT_PATH)
+  results:
+    - name: snapshotCreationComponent
+      description: Name of C
+  script: |
+    #!/usr/bin/env bash
+    set -eux
+
+    echo "Single Component mode? ${SINGLE_COMPONENT}"
+    if [ "${SINGLE_COMPONENT}" == "true" ]; then
```

Comment:
Why not handle this in the pipeline? If single mode, don't even run the task? Or does this make it easier to consume by EC as well?

Reply:
With the EC case, there is only 1 Task. We are using the StepAction to insert logic before the validate step.

Reply:
ok